File: front/src/app/core/models/parcela.ts
export interface Parcela {
  id: number;
  ownerId: number;
  ownerNombre: string;
  nombre: string;
  descripcion?: string;
  superficieHa: number;
  latitud: number;
  longitud: number;
  municipio?: string;
  provincia?: string;
  referenciaCatastral?: string;
  activa: boolean;
  createdAt: string;
}

export interface CrearParcelaRequest {
  nombre: string;
  descripcion?: string;
  superficieHa: number;
  latitud: number;
  longitud: number;
  municipio?: string;
  provincia?: string;
  referenciaCatastral?: string;
}

export interface ActualizarParcelaRequest {
  nombre?: string;
  descripcion?: string;
  superficieHa?: number;
  latitud?: number;
  longitud?: number;
  municipio?: string;
  provincia?: string;
  referenciaCatastral?: string;
}

File: back/src/main/java/com/tfg/agrogestion/domain/parcela/entity/Parcela.java
package com.tfg.agrogestion.domain.parcela.entity;

import com.tfg.agrogestion.domain.user.entity.Usuario;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedBy;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedBy;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "parcelas")
@EntityListeners(AuditingEntityListener.class)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Parcela {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id", nullable = false)
    private Usuario owner;

    @Column(nullable = false, length = 150)
    private String nombre;

    @Column(columnDefinition = "NVARCHAR(500)")
    private String descripcion;

    @Column(name = "superficie_ha", nullable = false, precision = 10, scale = 4)
    private BigDecimal superficieHa;

    @Column(nullable = false, precision = 10, scale = 7)
    private BigDecimal latitud;

    @Column(nullable = false, precision = 10, scale = 7)
    private BigDecimal longitud;

    @Column(length = 150)
    private String municipio;

    @Column(length = 100)
    private String provincia;

    @Column(name = "referencia_catastral", length = 50)
    private String referenciaCatastral;

    @Column(nullable = false)
    private Boolean activa;

    @CreatedDate
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @CreatedBy
    @Column(name = "created_by", length = 100)
    private String createdBy;

    @LastModifiedBy
    @Column(name = "updated_by", length = 100)
    private String updatedBy;
}